### ISSUE TITLE
Invalid type error when parsing metadata for packages

### ISSUE THREAD

**Issue by NorfairKing** [louib/nix2sbom#10]
I just wanted to try this out but got this:

```
$ nix run "https://flakehub.com/f/louib/nix2sbom/0.1.66.tar.gz" -
[00:00:00.000] (7f113f94a7c0) INFO   Getting the derivations from the current system.
[00:00:02.061] (7f113f94a7c0) INFO   Found 14271 derivations
[00:00:02.061] (7f113f94a7c0) INFO   Getting the metadata for packages in the Nix store
Error: "invalid type: map, expected a string at line 493167 column 10"
```

**Comment by louib** [comment]
@NorfairKing Thanks for reporting the issue!

This step is fetching the metadata for the packages in your store, which explains why I wasn't able to reproduce locally.
You can generate the metadata information for your Nix store with the following command:
```
nix-env -q -a --meta --json '.*' > packages-metadata.json
```
If you could send that over to me by email, I could investigate further. There shouldn't be anything sensitive in there, but I'm guessing email might make you feel more at ease than attaching it in this thread.

**Comment by louib** [comment]
@NorfairKing thanks for the additional details, I opened https://github.com/louib/nix2sbom/pull/11 with a failing unit test. I should be able to provide a fix soon. Looks like the issue is that the `maintainers` field is a list of lists :open_mouth:

**Comment by louib** [comment]
It looks like this package meta in particular is malformed, but was fixed in https://github.com/NixOS/nixpkgs/commit/f14b6f553a7721b963cf10048adf35d08d5d0253

**Comment by louib** [comment]
@NorfairKing I'm still pondering about how to best handle this derivation. In the meantime, [I added a `--metadata-path` option](https://github.com/louib/nix2sbom/pull/12) so that you can prune this specific derivation from the pre-generated metadata file. Assuming that the metadata file is called `packages-metadata.json`, you can now run the following command:
```
nix2sbom -f .# --metadata-path packages-metadata.json
```

**Comment by louib** [comment]
@NorfairKing I merged #11, which should fix this issue. Please give it another shot.
